src/main/java/org/junit/rules/RuleChain.java

```diff
@@ -77,8 +77,12 @@ private RuleChain(List<TestRule> rules) {
      *
      * @param enclosedRule the rule to enclose.
      * @return a new {@code RuleChain}.
+     * @throws IllegalArgumentException if {@code enclosedRule} is null
      */
     public RuleChain around(TestRule enclosedRule) {
+        if (enclosedRule == null) {
+            throw new IllegalArgumentException("The enclosed rule should not be null");
```

Comment:
We generally throw `NullPointerException` if someone passes in a `null` for a parameter that shouldn't be `null` (this is also what the Java Collections library does and what Effective Java recommends).